virttest/utils_net.py

```diff
@@ -4468,3 +4468,96 @@ def delete_linux_bridge_tmux(linux_bridge_name, iface_name=None, ignore_status=F
     else:
         cmd = 'ip link delete %s' % linux_bridge_name
     return utils_misc.cmd_status_output(cmd, shell=True, verbose=True, ignore_status=ignore_status)
+
+
+def check_filter_rules(ifname, bandwidth, expect_none=False):
+    """
+    Check bandwidth settings via 'tc filter' output
+
+    :param ifname: the interface name
+    :param bandwidth: the settings in the xml
+    :param expect_none: whether or not expect nothing in output,
+                        default to be False
+    :return: return True if get expected result, else return False
+    """
+    cmd = "tc -d filter show dev %s parent ffff:" % ifname
+    filter_output = process.run(cmd, shell=True).stdout_text
+    logging.debug("Bandwidth filter output: %s", filter_output)
+    if expect_none:
+        return not filter_output.strip()
+    if not filter_output.count("filter protocol all pref"):
+        logging.error("Can't find 'protocol all' settings in filter rules")
+        return False
+    filter_pattern = r".*police.*rate (\d+)(K?M?)bit burst (\d+)(K?M?)b.*"
+    tc_police = re.search(r"%s" % filter_pattern, filter_output, re.M)
+    if not tc_police:
+        logging.error("Can't find any filter policy")
+        return False
+    logging.debug("bandwidth from tc output:%s" % str(tc_police.groups()))
+    logging.debug("bandwidth from setting:%s" % str(bandwidth))
+    try:
+        if "average" in bandwidth:
+            if tc_police.group(2) == 'M':
+                tc_average = int(tc_police.group(1)) * 1000
+            else:
+                tc_average = int(tc_police.group(1))
+            assert tc_average == int(bandwidth["average"]) * 8
+        if "burst" in bandwidth:
+            if tc_police.group(4) == 'M':
+                tc_burst = int(tc_police.group(3)) * 1024
+            else:
+                tc_burst = int(tc_police.group(3))
+            assert tc_burst == int(bandwidth["burst"])
+    except Exception as e:
+        logging.debug("Can't get expected bandwidth from tc: %s", str(e))
```

Comment:
@yalzhang I don't think you'll get any useful info from `Exception as e`. I think you should update as you did in tp-libvirt pr. 
>>> try:
...  assert 4==5
... except Exception as e:
...  print('ddd')
...  print(e)
... 
ddd

>>> 

```suggestion
    except AssertionError:
        stacktrace.log_exc_info(sys.exc_info())
```

Reply:
Sigh, I always copy-paste without thinking. Thank you so much!